src/Paket.Core/Installation/InstallProcess.fs

```diff
@@ -293,20 +296,31 @@ let private applyBindingRedirects isFirstGroup createNewBindingFiles cleanBindin
                 librariesForPackage
                 |> Seq.choose(fun (library,redirects,profile) ->
                     try
-                        let assembly = Mono.Cecil.AssemblyDefinition.ReadAssembly(library.Path)
-                        Some (assembly, BindingRedirects.getPublicKeyToken assembly, assembly.MainModule.AssemblyReferences, redirects, profile)
+                        let assemblyName = AssemblyName.GetAssemblyName(library.Path)
+                        let publicKeyToken = AssemblyMetadata.getPublicKeyToken assemblyName
+                        let metadataName = AssemblyMetadata.AssemblyName assemblyName
+                        let references = 
+                            match AssemblyMetadata.getAssemblyReferences metadataName with
+                            | Trial.Pass list -> list
+                            | Trial.Warn(list, warn) -> 
```

Comment:
Where does `Trial.Warn` come from?

Reply:
Chessie – it seemed convenient to use here

Reply:
But I don't see it emitted, so this case can never happen, correct?

Reply:
AFAIK it can be produced in `getAssemblyReferences` by the `pass|fail –> collect` pipeline